vals_httpjson_test.go

```diff
@@ -0,0 +1,368 @@
+package vals
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	config2 "github.com/helmfile/vals/pkg/config"
+	"github.com/helmfile/vals/pkg/providers/httpjson"
+)
+
+var server *httptest.Server
+
+func setup() {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Define the JSON data
+		data := []map[string]interface{}{
+			{
+				"name": "chartify",
+				"id":   251296379,
+				"status": map[string]interface{}{
+					"database": map[string]interface{}{
+						"DBNodes": []string{
+							"chartify.database1.io",
+							"chartify.database2.io",
+							"chartify.database3.io",
+							"chartify.database4.io",
+							"chartify.database5.io",
+						},
+					},
+				},
+				"owner": map[string]interface{}{
+					"login": "helmfile",
+					"id":    8319146,
+				},
+			},
+			{
+				"name": "go-yaml",
+				"id":   597918420,
+				"status": map[string]interface{}{
+					"database": map[string]interface{}{
+						"DBNodes": []string{
+							"go-yaml.database1.io",
+							"go-yaml.database2.io",
+							"go-yaml.database3.io",
+							"go-yaml.database4.io",
+							"go-yaml.database5.io",
+						},
+					},
+				},
+				"owner": map[string]interface{}{
+					"login": "helmfile",
+					"id":    83191469,
+				},
+			},
+		}
+
+		// Encode the JSON data
+		jsonData, err := json.Marshal(data)
+		if err != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+
+		// Set the Content-Type header
+		w.Header().Set("Content-Type", "application/json")
+
+		// Write the JSON response
+		w.Write(jsonData)
+	})
+
+	// Create a test server (using any free port)
+	server = httptest.NewServer(handler)
+}
+
+func teardown() {
+	// Close the test server
+	server.Close()
+}
+
+func createProvider(providerPath string, inlineValue string, floatAsInt string) config2.MapConfig {
+	// Construct the configuration map with the provided values
+	config := map[string]interface{}{
+		"provider": map[string]interface{}{
+			"name":       "httpjson",
+			"path":       providerPath,
+			"floatAsInt": floatAsInt,
+			"insecure":   "true",
+		},
+		"inline": map[string]interface{}{
+			"value": inlineValue,
+		},
+	}
+	return config2.Map(config)
+}
+
+// nolint
```

Comment:
why nolint? I can help you to solve lint issue.

Reply:
remove this. then I will merge this PR.

Reply:
Removed nolint